phalcon/paginator/adapter/querybuilder.zep

```diff
@@ -191,11 +220,20 @@ class QueryBuilder extends Adapter
 
 		/**
 		 * Obtain the result of the total query
+		 * If we have having perform native count on temp table
 		 */
-		let result = totalQuery->execute(),
-			row = result->getFirst(),
-			rowcount = row ? intval(row->rowcount) : 0,
-			totalPages = intval(ceil(rowcount / limit));
+		if hasHaving {
+		    let sql = totalQuery->getSql();
+		    let db = totalBuilder->getDI()->get("db");
+		    let row = db->fetchOne(sprintf("SELECT COUNT(*) as rowcount FROM (%s) as T1", sql["sql"]), Db::FETCH_ASSOC, sql["bind"]),
```

Comment:
It looks cleaner imho than:

`"SELECT COUNT(*) as rowcount FROM (" . sql["sql"] . " as T1"`